File: string/14_LongestCommonPrefix/horizontal/Solution.py
class Solution:
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if (len(strs) == 0):
            return ""
        prefix = strs[0]
        for i in range(1, len(strs)):
            print(i)
            while(prefix not in strs[i][0:len(prefix)]):
                print(prefix)
                prefix = prefix[0: len(prefix) - 1]
                if (prefix == ""):
                    return "";
        return prefix

File: string/14_LongestCommonPrefix/horizontal/Solution.go
import "fmt"
func longestCommonPrefix(strs []string) string {
        if len(strs) == 0 {
                return ""
        }
        prefix := strs[0]
        for _, word := range strs {
            for !strings.HasPrefix(word, prefix) {
                        prefix = prefix[0 : len(prefix)-1]
                        if prefix == "" {
                                return ""
                        }
                }

        }
        return prefix
}